tests/src/main/java/com/hazelcast/simulator/tests/map/PagingPredicateTest.java

```diff
@@ -0,0 +1,149 @@
+    public boolean sequentialWorker = true;
+
+    private IMap<Integer, Employee> map;
+    private SqlPredicate innerPredicate;
+
+    @Setup
+    public void setUp(TestContext testContext) throws Exception {
+        HazelcastInstance targetInstance = testContext.getTargetInstance();
+        map = targetInstance.getMap(basename);
+        innerPredicate = new SqlPredicate(innerPredicateQuery);
+    }
+
+    @Warmup(global = true)
+    public void globalWarmup() {
+        if (useIndex) {
+            map.addIndex("salary", true);
+        }
+        initMapLoad();
+    }
+
+    @RunWithWorker
+    public AbstractMonotonicWorker createWorker() {
+        return sequentialWorker ? new SequentialWorker() : new RandomWorker();
+    }
+
+    private void initMapLoad() {
+        MapStreamer<Integer, Employee> streamer = MapStreamerFactory.getInstance(map);
+        for (int i = 0; i < keyCount; i++) {
+            Employee employee = new Employee(i);
+            streamer.pushEntry(employee.getId(), employee);
+        }
+        streamer.await();
+    }
+
+    private class SequentialWorker extends AbstractMonotonicWorker {
+        private PagingPredicate predicate;
+        private int predicateReusedCount;
+
+        @Override
+        protected void beforeRun() {
+            predicate = createNewPredicate();
+        }
+
+        @Override
+        protected void timeStep() {
+            createNewPredicateIfNeeded();
+            evaluatePredicate();
+            predicate.nextPage();
+        }
+
+        private void evaluatePredicate() {
+            map.entrySet(predicate);
+            predicateReusedCount++;
+        }
+
+        private void createNewPredicateIfNeeded() {
+            if (predicateReusedCount == maxPredicateReuseCount) {
+                predicate = createNewPredicate();
+                predicateReusedCount = 0;
+            }
+        }
+    }
+
+    private class RandomWorker extends AbstractMonotonicWorker {
+        private PagingPredicate predicate;
+        private int predicateReusedCount;
+
+        @Override
+        protected void beforeRun() {
+            predicate = createNewPredicate();
+        }
+
+        @Override
+        protected void timeStep() {
+            createNewPredicateIfNeeded();
+            goToRandomPage();
+            evaluatePredicate();
+        }
+
+        private void evaluatePredicate() {
+            map.entrySet(predicate);
+            predicateReusedCount++;
+        }
+
+        private void createNewPredicateIfNeeded() {
+            if (predicateReusedCount == maxPredicateReuseCount) {
+                predicate = createNewPredicate();
+                predicateReusedCount = 0;
+            }
+        }
+
+        private void goToRandomPage() {
+            int newPage = abs(randomInt(maxPage));
+            while (predicate.getPage() != newPage) {
+                if (predicate.getPage() > newPage) {
+                    predicate.previousPage();
```

Comment:
this is a local call. it will just increment or decrement an internal variable inside `PagingPredicate`